sequence_file_input and view::take don't work together

```c++
    sequence_file_input query_file{query_file_path, fields<field::ID, field::SEQ>{}};
    for (auto & [id, query] : query_file | view::take(30))
    {
        debug_stream << "id:\t\t" << id << '\n';
        debug_stream << "query:\t\t" << query << '\n';
    }
```
won't compile and outputs:

```
In file included from /seqan3/include/seqan3/range/view/take.hpp:50,
                 from /seqan3/include/seqan3/io/sequence_file/format_fasta.hpp:67,
                 from /seqan3/include/seqan3/io/sequence_file/input.hpp:63,
                 from /SeqAn3_read_mapper/mapper.cpp:2:
/seqan3/include/seqan3/range/detail/inherited_iterator_base.hpp: In instantiation of ‘class seqan3::detail::inherited_iterator_base<seqan3::detail::view_take<ranges::v3::iterator_range<seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > >, ranges::v3::default_sentinel>, false, false>::iterator_type, seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > > >’:
/seqan3/include/seqan3/range/view/take.hpp:88:11:   required from ‘class seqan3::detail::view_take<ranges::v3::iterator_range<seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > >, ranges::v3::default_sentinel>, false, false>::iterator_type’
/SeqAn3_read_mapper/mapper.cpp:75:57:   required from here
/seqan3/include/seqan3/range/detail/inherited_iterator_base.hpp:83:89: error: no type named ‘difference_type’ in ‘struct std::iterator_traits<seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > > >’
     using difference_type       = typename std::iterator_traits<base_t>::difference_type;
                                                                                         ^
/seqan3/include/seqan3/range/detail/inherited_iterator_base.hpp:84:84: error: no type named ‘value_type’ in ‘struct std::iterator_traits<seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > > >’
     using value_type            = typename std::iterator_traits<base_t>::value_type;
                                                                                    ^
/seqan3/include/seqan3/range/detail/inherited_iterator_base.hpp:85:83: error: no type named ‘reference’ in ‘struct std::iterator_traits<seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > > >’
     using reference             = typename std::iterator_traits<base_t>::reference;
                                                                                   ^
/seqan3/include/seqan3/range/detail/inherited_iterator_base.hpp:86:81: error: no type named ‘pointer’ in ‘struct std::iterator_traits<seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > > >’
     using pointer               = typename std::iterator_traits<base_t>::pointer;
                                                                                 ^
/seqan3/include/seqan3/range/detail/inherited_iterator_base.hpp:87:91: error: no type named ‘iterator_category’ in ‘struct std::iterator_traits<seqan3::detail::in_file_iterator<seqan3::sequence_file_input<seqan3::sequence_file_input_default_traits_dna, seqan3::fields<(seqan3::field)1, (seqan3::field)0>, meta::v1::list<seqan3::sequence_file_format_fasta, seqan3::sequence_file_format_fastq>, std::basic_ifstream<char> > > >’
     using iterator_category     = typename std::iterator_traits<base_t>::iterator_category;
                                                                                           ^
```

I'm not sure if this is intended or if this is a bug.

--- a/include/seqan3/io/detail/in_file_iterator.hpp
+++ b/include/seqan3/io/detail/in_file_iterator.hpp
@@ -79,6 +79,7 @@ public:
     using const_reference   = typename file_type::reference;
     using size_type         = typename file_type::size_type;
     using difference_type   = typename file_type::difference_type;
+    using pointer           = typename file_type::value_type *;
     //!\brief Tag this class as an input iterator.
     using iterator_category = std::input_iterator_tag;
     //!\}

--- a/include/seqan3/io/detail/out_file_iterator.hpp
+++ b/include/seqan3/io/detail/out_file_iterator.hpp
@@ -85,6 +85,7 @@ public:
     using const_reference   = void;
     using size_type         = void;
     using difference_type   = std::ptrdiff_t;
+    using pointer           = void *;
     //!\brief Tag this class as an input access iterator.
     using iterator_category = std::output_iterator_tag;
     //!\}

--- a/include/seqan3/range/view/detail.hpp
+++ b/include/seqan3/range/view/detail.hpp
@@ -61,6 +61,26 @@ struct view_base : public ranges::view_base
 {};
 
 // ============================================================================
+//  size_type_t_or_void
+// ============================================================================
+
+//!\brief Transformation trait that resolves to seqan3::size_type_t if possible and void otherwise. [Default Overload]
+template <typename t>
+struct size_type_t_or_void
+{
+    //!\brief The default is to expose `void`.
+    using type = void;
+};
+
+//!\brief Transformation trait that resolves to seqan3::size_type_t if possible and void otherwise. [Specialisation]
+template <std::ranges::SizedRange t>
+struct size_type_t_or_void<t>
+{
+    //!\brief For sized ranges the seqan3::size_type_t is exposed.
+    using type = size_type_t<t>;
+};
+
+// ============================================================================
 //  pipable_adaptor_base
 // ============================================================================
 

--- a/include/seqan3/range/view/persist.hpp
+++ b/include/seqan3/range/view/persist.hpp
@@ -87,7 +87,7 @@ public:
     //!\brief The value_type (which equals the reference_type with any references removed).
     using value_type        = value_type_t<urng_t>;
     //!\brief The size_type is `size_t` if the the view is exact, otherwise void.
-    using size_type         = std::conditional_t<std::ranges::SizedRange<urng_t>, size_type_t<urng_t>, void>;
+    using size_type         = typename detail::size_type_t_or_void<urng_t>::type;
     //!\brief A signed integer type, usually std::ptrdiff_t.
     using difference_type   = difference_type_t<urng_t>;
     //!\brief The iterator type of this view (a random access iterator).

--- a/test/unit/io/sequence_file/CMakeLists.txt
+++ b/test/unit/io/sequence_file/CMakeLists.txt
@@ -1,4 +1,5 @@
 seqan3_test(sequence_file_input_test.cpp)
+seqan3_test(sequence_file_integration_test.cpp)
 seqan3_test(sequence_file_output_test.cpp)
 seqan3_test(sequence_file_format_fasta_test.cpp)
 seqan3_test(sequence_file_format_fastq_test.cpp)

--- a/test/unit/io/sequence_file/sequence_file_integration_test.cpp
+++ b/test/unit/io/sequence_file/sequence_file_integration_test.cpp
@@ -1,0 +1,165 @@
+// ==========================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ==========================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert, FU Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <seqan3/io/sequence_file/output.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/range/view/persist.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+#include <seqan3/std/iterator>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+TEST(rows, assign_sequence_files)
+{
+    std::string const input
+    {
+        ">TEST 1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN\n\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    std::string const output_comp
+    {
+        "> TEST 1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    sequence_file_input fin{std::istringstream{input}, sequence_file_format_fasta{}};
+    sequence_file_output fout{std::ostringstream{}, sequence_file_format_fasta{}};
+    fout.options.fasta_letters_per_line = 0;
+
+    fout = fin;
+
+    fout.get_stream().flush();
+    EXPECT_EQ(fout.get_stream().str(), output_comp);
+}
+
+TEST(integration, assign_sequence_file_pipes)
+{
+    std::string const input
+    {
+        "> TEST1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    // valid without assignment?
+    sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} |
+        sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
+
+    // valid with assignment and check contents
+    auto fout = sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} |
+                sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
+
+    fout.get_stream().flush();
+    EXPECT_EQ(fout.get_stream().str(), input);
+}
+
+TEST(integration, view)
+{
+    std::string const input
+    {
+        "> TEST1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    std::string const output
+    {
+        "> TEST1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+    };
+
+    // valid without assignment?
+    sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} | view::persist | view::take(2) |
+        sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
+
+    // valid with assignment and check contents
+    auto fout = sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}}
+              | view::persist
+              | view::take(2)
+              | sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
+
+    fout.get_stream().flush();
+    EXPECT_EQ(fout.get_stream().str(), output);
+}
+
+TEST(integration, convert_fastq_to_fasta)
+{
+    std::string const fastq_in
+    {
+    "@ID1\n"
+    "ACGTT\n"
+    "+\n"
+    "!##$%\n"
+    "@ID2\n"
+    "TATTA\n"
+    "+\n"
+    ",BDEB\n"
+    };
+
+    std::string const fasta_out
+    {
+        "> ID1\n"
+        "ACGTT\n"
+        "> ID2\n"
+        "TATTA\n"
+    };
+
+    auto fout = sequence_file_input{std::istringstream{fastq_in}, sequence_file_format_fastq{}} |
+                sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
+    fout.get_stream().flush();
+    EXPECT_EQ(fout.get_stream().str(), fasta_out);
+}

--- a/test/unit/io/sequence_file/sequence_file_output_test.cpp
+++ b/test/unit/io/sequence_file/sequence_file_output_test.cpp
@@ -40,7 +40,6 @@
 #include <range/v3/view/filter.hpp>
 
 #include <seqan3/io/sequence_file/output.hpp>
-#include <seqan3/io/sequence_file/input.hpp>
 #include <seqan3/range/view/convert.hpp>
 #include <seqan3/range/view/to_char.hpp>
 #include <seqan3/test/tmp_filename.hpp>
@@ -57,7 +56,6 @@ TEST(sequence_file_output_iterator, concepts)
     EXPECT_TRUE((std::OutputIterator<it_t, std::tuple<std::string, std::string>>));
     EXPECT_TRUE((std::Sentinel<sen_t, it_t>));
 }
-
 
 std::vector<dna5_vector> seqs
 {
@@ -386,75 +384,6 @@ TEST(rows, assign_range_of_tuples)
         range.emplace_back(seqs[i], ids[i]);
 
     assign_impl(range);
-}
-
-TEST(rows, assign_sequence_file_input)
-{
-    std::string const input // differs from output above by formatting
-    {
-        ">TEST 1\n"
-        "ACGT\n"
-        "> Test2\n"
-        "AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN\n\n"
-        "> Test3\n"
-        "GGAGTATAATATATATATATATAT\n"
-    };
-
-    sequence_file_input fin{std::istringstream{input}, sequence_file_format_fasta{}};
-
-    assign_impl(fin);
-}
-
-TEST(rows, assign_sequence_file_pipes)
-{
-    std::string const input
-    {
-        "> TEST1\n"
-        "ACGT\n"
-        "> Test2\n"
-        "AGGCTGNAGGCTGAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
-        "> Test3\n"
-        "GGAGTATAATATATATATATATAT\n"
-    };
-
-    // valid without assignment?
-    sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} |
-        sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
-
-    // valid with assignment and check contents
-    auto fout = sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} |
-                sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
-
-    fout.get_stream().flush();
-    EXPECT_EQ(fout.get_stream().str(), input);
-}
-
-TEST(rows, convert_fastq_to_fasta)
-{
-    std::string const fastq_in
-    {
-    "@ID1\n"
-    "ACGTT\n"
-    "+\n"
-    "!##$%\n"
-    "@ID2\n"
-    "TATTA\n"
-    "+\n"
-    ",BDEB\n"
-    };
-
-    std::string const fasta_out
-    {
-        "> ID1\n"
-        "ACGTT\n"
-        "> ID2\n"
-        "TATTA\n"
-    };
-
-    auto fout = sequence_file_input{std::istringstream{fastq_in}, sequence_file_format_fastq{}} |
-                sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
-    fout.get_stream().flush();
-    EXPECT_EQ(fout.get_stream().str(), fasta_out);
 }
 
 // ----------------------------------------------------------------------------

